homeassistant/components/google/calendar.py

```diff
@@ -310,12 +332,15 @@ def __init__(
     ) -> None:
         """Create the Calendar event device."""
         super().__init__(coordinator)
+        _LOGGER.debug("entity_description.entity_id=%s", entity_description.entity_id)
+        _LOGGER.debug("entity_description=%s", entity_description)
         self.calendar_id = calendar_id
         self.entity_description = entity_description
         self._ignore_availability = entity_description.ignore_availability
         self._offset = entity_description.offset
         self._event: CalendarEvent | None = None
-        self.entity_id = entity_description.entity_id
+        if entity_description.entity_id:
+            self.entity_id = entity_description.entity_id
```

Comment:
Why do we let the entity decide the entity_id and why don't we let HA generate it themselves?

Reply:
It's historical only, for entities created from yaml configuration. I've been assuming this would be a painful breaking change to make. My best thought was to stop doing it for this new entity type since it can use new style names with translations etc

Reply:
If the entity has a unique_id you can remove it, since the used entity_id is stored in the entity registry, that's why naming changes (like the whole has_entity_name migration) is not breaking as all the entity_ids will stay consistent

Reply:
OK great. Not all entities made by this integration have a unique id, but many of them do.  The ones that do not are the ones that are named based on some yaml setting. How about in a follow up PR I can stop setting the entity_id on the entity description for those cases? This change, to reduce the number of cases when entity ids are explicitly set, would still be in line with that and its still needed to set it sometimes.